File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n=height.size();
        int left=0;
        int right=n-1;
        int max_area=INT_MIN;
        while(left<=right)
        {
            int pre_area=min(height[left],height[right])*(right-left);
            max_area=max(pre_area,max_area);
            if(height[left]<height[right])
            left++;
            else
            right--;


        }
        return max_area;
        
    }
};

File: 11-container-with-most-water/container-with-most-water.py
import math
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        n=len(height)
        l=0
        r=n-1
        i=0
        area=-1

        while(l<r):
            h=min(height[l],height[r])
            ind=r-l
            area1=h*ind
            area=max(area1,area)
            if height[l]<height[r]:
                l+=1
            else:
                r-=1    

        return area    
        